java-squid/src/main/java/org/sonar/java/model/expression/NewArrayTreeImpl.java

```diff
@@ -93,11 +119,15 @@ public void accept(TreeVisitor visitor) {
 
   @Override
   public Iterator<Tree> childrenIterator() {
-    return Iterators.concat(
-      Iterators.singletonIterator(type),
-      dimensions.iterator(),
-      initializers.iterator()
-      );
+    ImmutableList.Builder<Tree> iteratorBuilder = ImmutableList.<Tree>builder();
+    addIfNotNull(iteratorBuilder, newKeyword);
+    addIfNotNull(iteratorBuilder, type);
+    // TODO SONARJAVA-547 Brackets of dimensions are missing
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)